### DISCUSSION TITLE
Several extension connects to localhost instead of code-server

### DISCUSSION THREAD

**Discussion by tophrr** [coder/code-server#3205] [Q&A] [answered]
I have just installed code-server on a VM (another host, not local) the other night, when I was syncing my extensions from my desktop VSCode I noticed several extensions not working. I set up caddy in another VM to reverse proxy to code-server from a domain. I cannot connect to the extension local service because it is in localhost, but I am accessing code-server through a subdomain on a reverse proxy. For example, the Tabnine extension has a local http service at port 5555 for its setup. When I access the setup page, my browser connects to 127.0.0.1:5555 instead of code-server which results in a blank page. Is there any way to change it to access <code-server domain>/proxy/5555 instead? Or am I missing a step here?
Here is my config file
```
bind-addr: 172.16.0.4:8080 # reverse proxy subnet
auth: password
password: redacted
cert: false
```

**Comment by code-asher** [comment, marked as answer]
I'm not familiar with the Tabnine extension but there might be a setting to change the URL. I think most extensions assume they are running locally so they will not have this option.

Does it work if you manually go to `/proxy/5555` to complete setup?

If not, some workarounds I can think of are:

1. Edit the extension code so it uses your domain instead of 127.0.0.1.
2. Forward the port from the remote machine to your local machine.
